Gutenberg: Add and use jetpack_register_block_type() wrapper

ockham opened the pull request:
This includes a `function_exists()` check to avoid usage of `register_block_type` in older WordPress versions that don't yet have it, leading to fatals (which my recent #11212 caused -- very sorry!)

Some convo (with @dereksmart and @jeherve) at p1549647686423800-slack-jetpack-gutenberg

#### Changes proposed in this Pull Request:
<!--- Explain what functional changes your PR includes -->

* Add and use `jetpack_register_block_type()` wrapper instead of `register_block_type()`

#### Testing instructions:

- Verify that blocks are still registered and available
- On a test site running WP 4.9.9, verify that no fatals occur.
- Grep for `register_block_type` to make sure we're not calling it directly anymore.

#### Follow-up

As a follow-up, we should modify `tests/php/test_class.jetpack-gutenberg.php` (and `Jetpack_Gutenberg`) so it won't skip tests on a WP installation without `register_block_type` to make sure that `Jetpack_Gutenberg` doesn't fatal.

#### Proposed changelog entry for your changes:
<!-- Please do not leave this empty. If no changelog entry needed, state as such. -->

* Add and use `jetpack_register_block_type()` wrapper instead of `register_block_type()`

enejb commented on class.jetpack-gutenberg.php:
```diff
@@ -6,10 +6,29 @@
+ *
+ * @return WP_Block_Type|false The registered block type on success, or false on failure.
+ */
+function safe_register_block_type( $slug, $args = array() ) {
```
Lets prefix this with jetpack :)

jeherve replied:
Took care of that 👍

jeherve commented on tests/php/sync/test_class.jetpack-sync-callables.php:
```diff
@@ -55,7 +55,7 @@ public function test_sync_callable_whitelist() {
 		if ( function_exists( 'register_block_type' ) ) {
```
Could we remove the check here now?

ockham replied:
I'll see if that works with WP<5.0 without requiring modifications to the `Jetpack_Gutenberg` class -- otherwise I'll defer to a follow-up PR if that's okay.

ockham replied:
Works: a4e71b18a

jeherve reviewed (dismissed):
This works well for me. I only have one minor comment.

ockham commented:
Thanks for adding the `jetpack_` prefix, @jeherve -- makes sense to avoid collisions with other plugins that way.

I know I'm the one who suggested the `safe_` infix, but I'm having second thoughts now - `jetpack_safe_register_block_type` is quite a mouthful :sweat_smile: Should we drop the `safe_` after all? (`jetpack_register_block_type` won't collide with the other now-deprecated previous function which was fortunately called `jetpack_register_block`.)

jeherve commented:
> Should we drop the `safe_` after all?

Yeah, you're right. It seems safe without it. Let's drop it.

ockham commented:
Changed the function name to `jetpack_register_block_type` and removed the now-obsolete `function_exists()` check from `tests/php/sync/test_class.jetpack-sync-callables.php`.

ockham commented:
Putting on hold to get #11312 in first.

jeherve commented:
This should now be good to rebase. 👍

ockham committed:
Gutenberg: Add and use jetpack_register_block_type() wrapper

ockham commented:
Rebased.

jeherve reviewed (approved):
Looking good. Merge when ready.
